core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -94,68 +190,65 @@ class Partition(val topicPartition: TopicPartition,
   private var controllerEpoch: Int = KafkaController.InitialControllerEpoch
   this.logIdent = s"[Partition $topicPartition broker=$localBrokerId] "
 
-  private def isReplicaLocal(replicaId: Int) : Boolean = replicaId == localBrokerId || replicaId == Request.FutureLocalReplicaId
+  private def isReplicaLocal(replicaId: Int): Boolean = replicaId == localBrokerId || replicaId == Request.FutureLocalReplicaId
 
   private val tags = Map("topic" -> topic, "partition" -> partitionId.toString)
 
-  // Do not create metrics if this partition is ReplicaManager.OfflinePartition
-  if (!isOffline) {
```

Comment:
So we're able to remove the `isOffline` flag here due to the new `OfflinePartition` in `ReplicaManager` (which does not include a `Partition` instance)?

Reply:
Yes, this is something I've been meaning to do for quite a while. The sentinel we had before is just a bad practice.